src/node/node_state.h

```diff
@@ -119,8 +119,9 @@ namespace ccf
     static constexpr NodeId invalid_node_id = -1;
     NodeId self = invalid_node_id;
     tls::KeyPairPtr node_sign_kp;
-    tls::KeyPairPtr node_encrypt_kp;
+    std::shared_ptr<tls::KeyPair_mbedTLS> node_encrypt_kp;
```

Comment:
I've probably missed it but should we define the following and have it configurable from `cmake` so that we can easily switch between provider?

```C++
#ifdef CRYPTO_PROVIDER_IS_MBEDTLS
    using KeyPair = KeyPair_mbedTLS;
#else
    using KeyPair = KeyPair_openSSL;
#endif
```

Reply:
Sure, makes sense!